### ISSUE TITLE
Write documentation on guidelines / philosophy behind reusable effects

### ISSUE THREAD

**Issue by isovector** [isovector/polysemy#28]
Maybe a blog post?

**Comment by rabbitonweb** [comment]
Just so community is aware: I'm building two examples of polysemy usage:

1. ugly Haskell -> MTL -> polysem 

This will show why you want to write polymorphic code, how to do it with old MTL approach and then how to refactor it to polysemy

2. benefits of polysemy

This will show what are the true benefits of using code with polysemy. We will start from higher abstraction: with a user story that uses business language and then show how we can decompose the complexity, by:
a) defining different level DSL for different parts of business lingua 
b) defining lower-level DSL and showing how we can interpreter higher level language to lower level language

Both of those examples will come as blog posts and can be considered Quickstart/Tutorials to `polysemy`

So TL;DR: 
- I'm on it
- @isovector is aware that I'm on it. 
- This comment here is for the rest of the community to know that I'm working on it, so keep your fingers crossed :)

**Comment by thma** [comment]
I have written a tutorial that demonstrates how to 
implement a Servant REST application using Polysemy.

I've put much effort into achieving a clean separation 
of concerns between the different application layers.

I've organized the code according to the
the clean architecture guidelines.

One of the benefits of this approach is that
isolated components tests on each layer become super easy.

You'll find the tutorial and the source code here:

https://github.com/thma/PolysemyCleanArchitecture#readme

Feel free to use or to reference to any of this material 
if it fits your documentation purposes.

**Comment by isovector** [comment]
@thma this is amazing, thank you!
